js/router.js

```diff
@@ -1,15 +1,147 @@
+import $ from 'jquery';
 import { Router } from 'backbone';
+import { getGuid } from './utils';
+import { getPageContainer } from './utils/selectors';
+import UserPage from './views/UserPage';
+import TransactionsPage from './views/TransactionsPage';
+import TemplateOnly from './views/TemplateOnly';
 
 export default class ObRouter extends Router {
-  constructor() {
+  constructor(options = {}) {
+    super(options);
+    this.options = options;
+
+    // Temporarily requiring a dummy 'users' collection. Eventually, we'll
+    // be getting this data via fetches to the server
+    if (!options.usersCl) {
+      throw new Error('Need me some dummy users!');
+    }
+
+    this.usersCl = options.usersCl;
+
+    if (!options.pageNavVw) {
+      throw new Error('Please provide the app\'s pageNavVw.');
+    }
+
+    this.pageNavVw = options.pageNavVw;
+
     const routes = [
-      // ['home/:state(/:searchText)', 'home'],
-      // ['userPage/:userID(/:state)(/:itemHash)(/:skipNSFWmodal)', 'userPage'],
-      // [/^@([^\/]+)(.*)$/, 'userPageViaHandle'],
-      // ['transactions/:state(/:orderID)(/:tabState)', 'transactions'],
-      // ['settings/:state', 'settings'],
+      [/^@([^\/]+)[\/]?([^\/]*)[\/]?([^\/]*)[\/]?([^\/]*)$/, 'userViaHandle'],
```

Comment:
You could use regular expressions in routes, which are very powerful, or use the simpler syntax (e.g. in the transactions routes below), but you can't mix the two.

In the regular expression syntax, `[\/]?([^\/]*)` is more or less equivalent to `(:optional-path-frag)`